plugin/storage/integration/integration_test.go

```diff
@@ -0,0 +1,288 @@
+		}
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestFindTraces(t *testing.T) {
+	t.Log("Testing FindTraces ...")
+	t.Log("\t(Note: all cases include ServiceName + StartTime range)")
+	queries, err := getQueries()
+	require.NoError(t, err)
+	for _, query := range queries {
+		t.Logf("\t\t* Query case: + %s", query.Caption)
+		s.integrationTestFindTracesByQuery(t, query.Query, query.ExpectedTraces)
+	}
+}
+
+func (s *StorageIntegration) integrationTestFindTracesByQuery(t *testing.T, query *spanstore.TraceQueryParameters, expectedTraces []int) {
+	traces, err := getTraceFixtures()
+	require.NoError(t, err)
+	expected := getSubsetOfTraces(traces, expectedTraces)
+	require.NoError(t, s.writeTraces(expected))
+	require.NoError(t, s.refresh())
+
+	iterations := 10 * timeOut
+	var found bool
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf("Waiting for ES to update documents, iteration %d out of %d", i+1, iterations))
+		actual, err := s.reader.FindTraces(query)
+		require.NoError(t, err)
+
+		if len(actual) == query.NumTraces {
+			found = true
+			break
+		}
+		if found = err == nil && tracesMatch(actual, expected); found {
+			CompareListOfModelTraces(t, expected, actual)
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	assert.True(t, found)
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) writeTraces(traces []*model.Trace) error {
+	for _, trace := range traces {
+		if err := s.writeTrace(trace); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *StorageIntegration) writeTrace(trace *model.Trace) error {
+	for _, span := range trace.Spans {
+		if err := s.writer.WriteSpan(span); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getSubsetOfTraces(traces []*model.Trace, expectedTraces []int) []*model.Trace {
+	retTraces := make([]*model.Trace, len(expectedTraces))
+	for i, traceNum := range expectedTraces {
+		retTraces[i] = traces[traceNum-1]
+	}
+	return retTraces
+}
+
+func getTraceFixtures() ([]*model.Trace, error) {
+	traces := make([]*model.Trace, numOfTraceFixtures)
+	for i := 0; i < numOfTraceFixtures; i++ {
+		trace, err := getTraceFixture(i + 1)
+		if err != nil {
+			return nil, err
+		}
+		traces[i] = trace
+	}
+	return traces, nil
+}
+
+func getTraceFixture(i int) (*model.Trace, error) {
+	var trace model.Trace
+	fileName := fmt.Sprintf("fixtures/traces/trace_%02d.json", i)
+	err := getFixture(fileName, &trace)
+	if err != nil {
+		return nil, err
+	}
+	return &trace, nil
+}
+
+func getQueries() ([]*QueryFixtures, error) {
+	var queries []*QueryFixtures
+	err := getFixture("fixtures/queries.json", &queries)
+	if err != nil {
+		return nil, err
+	}
+	return queries, nil
+}
+
+func getFixture(path string, object interface{}) error {
+	inStr, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(normalizeTime(inStr), object)
+	return err
+}
+
+func normalizeTime(json []byte) []byte {
```

Comment:
correctTime? I think there's a better function name out there, we just have to find it. Also add a comment why we need this